index.js

```diff
@@ -132,53 +162,61 @@ function nowServing(line) {
   if (!line.length) {
     return "There is nobody waiting to be served!"
   }
+  takeATicketNumber(line)
   return `Currently serving ${line.shift()}.`
 }
 
+function takeATicketNumber(line) {
+  line.push(line.length + 1)
+  return `Welcome. You are ticket number ${line.length}` 
+}
```

Comment:
Very close! For this question, you'd want to declare a `counter` variable outside of your function, and then inside of the function, increment the `counter` and pushing that new number to the line.